client-data/tools/document/document.js

```diff
@@ -3,33 +3,47 @@
 
 // This isn't an HTML5 canvas, it's an old svg hack, (the code is _that_ old!)
 
-    var xlinkNS = "http://www.w3.org/1999/xlink";
-    var imgCount = 1;
+    const xlinkNS = "http://www.w3.org/1999/xlink";
+    let imgCount = 1;
+
     function onstart() {
         const fileInput = document.createElement("input");
         fileInput.type = "file";
         fileInput.accept = "image/*";
         fileInput.click();
         fileInput.addEventListener("change", () => {
-            var reader = new FileReader();
+            const reader = new FileReader();
             reader.readAsDataURL(fileInput.files[0]);
 
             reader.onload = function (e) {
-                var image = new Image();
+                const image = new Image();
                 image.src = e.target.result;
                 image.onload = function () {
 
                     var uid = Tools.generateUID("doc"); // doc for document
-                    console.log(image.src.toString().length);
 
-                    var msg = {
+                    // File size as data url, approximately 1/3 larger than as bytestream
+                    //TODO: internationalization
+                    let size = image.src.toString().length;
+                    if (size > 1048576) { //TODO: get correct size from config
```

Comment:
@lovasoa what would be the best way to access `config.MAX_DOCUMENT_SIZE` from here? I don't want to expose all values set in the config here.

Reply:
We could reuse the same mechanism we have for translations to pass the configuration to the clients.